tests/unit/constraints/test_tabular.py

```diff
@@ -431,44 +431,215 @@ def test___init___strict_false(self):
+        assert instance.constraint_columns == ('a', 'b')
 
-    def test___init___all_parameters_passed(self):
+    def test___init___high_is_scalar(self):
         """Test the ``GreaterThan.__init__`` method.
 
-        The passed arguments should be stored as attributes.
+        The passed arguments should be stored as attributes. Make sure ``scalar``
+        is set to ``'high'``.
 
         Input:
         - low = 'a'
-        - high = 'b'
+        - high = 0
         - strict = True
         - drop = 'high'
-        - high_is_scalar = True
-        - low_is_scalar = False
+        - scalar = 'high'
         Side effects:
         - instance._low == 'a'
-        - instance._high == 'b'
-        - instance._stric == True
+        - instance._high == 0
+        - instance._strict == True
         - instance._drop = 'high'
-        - instance._high_is_scalar = True
-        - instance._low_is_scalar = False
+        - instance._scalar == 'high'
         """
         # Run
-        instance = GreaterThan(low='a', high='b', strict=True, drop='high',
-                               high_is_scalar=True, low_is_scalar=False)
+        instance = GreaterThan(low='a', high=0, strict=True, drop='high', scalar='high')
 
         # Asserts
-        assert instance._low == 'a'
-        assert instance._high == 'b'
+        assert instance._low == ['a']
+        assert instance._high == 0
         assert instance._strict is True
-        assert instance._high_is_scalar is True
-        assert instance._low_is_scalar is False
+        assert instance._scalar == 'high'
         assert instance._drop == 'high'
+        assert instance.constraint_columns == ('a',)
+
+    def test___init___low_is_scalar(self):
+        """Test the ``GreaterThan.__init__`` method.
+
+        The passed arguments should be stored as attributes. Make sure ``scalar``
+        is set to ``'high'``.
+
+        Input:
+        - low = 0
+        - high = 'a'
+        - strict = True
+        - drop = 'low'
+        - scalar = 'low'
+        Side effects:
+        - instance._low == 0
+        - instance._high == 'a'
+        - instance._stric == True
+        - instance._drop = 'low'
+        - instance._scalar == 'low'
+        """
+        # Run
+        instance = GreaterThan(low=0, high='a', strict=True, drop='low', scalar='low')
+
+        # Asserts
+        assert instance._low == 0
+        assert instance._high == ['a']
+        assert instance._strict is True
+        assert instance._scalar == 'low'
+        assert instance._drop == 'low'
+        assert instance.constraint_columns == ('a',)
+
+    def test___init___high_low_both_list_error(self):
+        """Test the ``GreaterThan.__init__`` method.
+
+        Make sure to raise an error when both ``high`` and ``low`` are lists.
+
+        Input:
+        - low = ['a']
+        - high = ['b']
+        """
+        # Run / Assert
+        with pytest.raises(ValueError):
+            GreaterThan(low=['a'], high=['b'])
```

Comment:
Hm. So technically this should be valid because we're only passing in one column for high and one column for low right? 

In our validation we could check 
```
if isinstance(low, list):
    if len(low) > 1:
        raise ValueError('`low` cannot be multiple columns when `scalar` is `None`.')
    low = low[0]
elif isinstance(high, list):
    if len(high) > 1:
        raise ValueError('`high` cannot be a list when `scalar` is `None`.')
    high = high[0]
```

Reply:
I think this is now outdated in a sense that we allow `GreaterThan(low=['a'], high=['b'])`. but in general, internally we are dealing with slices of dataframes so all `low` and `high` values need to be stored in a list (except if they are scalar).